scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/component/ComponentDefinitionWithImplementation.scala

```diff
@@ -33,7 +33,7 @@ trait ComponentDefinitionWithImplementation extends ObjectOperatingOnTypes {
 
   protected def uiDefinition: ComponentUiDefinition
 
-  final def label: String = {
+  final def labelOrDefault: String = {
```

Comment:
1. I think that following the other methods approach, we can still name it `label` as it was before
2. `ComponentUiDefinition` is the place where we should do this fallback. Take a look that in that place everything is well known, determined based on config and other things. BTW you won't need to do any hacking with fragments than

Reply:
This `Ui` in `ComponentUiDefinition` means that we create an object that represents the situation that will be visible in UI, whereas `ComponentDefinition` represents properties of component declared by developer of component extension.